tests/test_rigid_physics.py

```diff
@@ -1293,9 +1293,11 @@ def test_set_root_pose(batch_fixed_verts, relative, show_viewer, tol):
         ):
             pos_zero = torch.tensor(pos_zero, device=gs.device, dtype=gs.tc_float)
             euler_zero = torch.deg2rad(torch.tensor(euler_zero, dtype=gs.tc_float))
+            quat_zero = gu.xyz_to_quat(euler_zero, rpy=True)
             assert_allclose(entity.get_pos(), pos_zero, tol=tol)
-            euler = gu.quat_to_xyz(entity.get_quat(), rpy=True)
-            assert_allclose(euler, euler_zero, tol=5e-4)
+            # Use quaternion for comparison to avoid gymbal lock issue in euler angles
+            quat = entity.get_quat()
+            assert_allclose(quat, quat_zero, tol=tol)
```

Comment:
Quaternion are not uniquely defined. The best approach would be:
```python
assert np.linalg.norm(gu.quat_to_rotvec(gu.transform_quat_by_quat(gu.inv_quat(quat), quat_zero))) < tol
```
or even better:
```python
assert_allclose(gu.quat_to_rotvec(gu.transform_quat_by_quat(gu.inv_quat(quat), quat_zero)), 0.0, tol=tol)
```